db/util_testing.go

```diff
@@ -0,0 +1,78 @@
+package db
+
+import (
+	"math"
+
+	"github.com/couchbase/gocb"
+	"github.com/couchbase/sync_gateway/base"
+)
+
+// Workaround SG #3570 by doing a polling loop until the star channel query returns 0 results.
+// Uses the star channel index as a proxy to indicate that _all_ indexes are empty (which might not be true)
+func WaitForIndexEmpty(bucket *base.CouchbaseBucketGoCB, bucketSpec base.BucketSpec) error {
+
+	retryWorker := func() (shouldRetry bool, err error, value interface{}) {
+
+		var results gocb.QueryResults
+
+		// Create the star channel query
+		starChannelQueryStatement := replaceSyncTokensQuery(QueryStarChannel.statement, bucketSpec.UseXattrs)
+		params := map[string]interface{}{}
+		params[QueryParamStartSeq] = 0
+		params[QueryParamEndSeq] = math.MaxInt64
+
+		// Execute the query
+		results, err = bucket.Query(starChannelQueryStatement, params, gocb.RequestPlus, true)
```

Comment:
Appending `LIMIT 1` to starChannelQueryStatement would be more efficient in cases where there's a lot of data in the index pre-flush.  Probably not a huge difference for typical test bucket data volumes, but since we're trying to streamline this as much as possible it won't hurt.

Reply:
Fixed in 000708f6cd889a112c3fd6e06c49955301ff3297